src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,56 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private int size;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ITEMS_NUMBER];
+        values = (V[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getKeyIndex(key);
+        if (index != -1) { //если такой key есть в массиве keys
+            values[index] = value; //то перезаписываем его value
+            if (index >= size) {
+                size++;
+            }
+        } else { //если нет
+            keys[size] = key; //то добавляем новую запись key & value
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int i = getKeyIndex(key);
+        if (i != -1) {
+            return values[i];
+        }
+        return null;//прийдется вернуть null хоть это и bad practice но так требуют тесты
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    private int getKeyIndex(K key) {
+        if (size != 0) {
+            int i = -1;
+            for (K element : keys) {
+                i++;
+                if ((element != null && element.equals(key)) || (element == null && key == null)) {
+                    return i;
+                }
+            }
+        }
         return -1;
```

Comment:
```suggestion
        for (int i = 0; i < index; i++) {
            if (key == null && keys[i] == null || key != null && key.equals(keys[i])) {
                return i;
            }
        }
        return -1;
```

реализация этого метода выглядит немного проще, если использовать обычный цикл `for`